### DISCUSSION TITLE
How to keep  viewer window open?

### DISCUSSION THREAD

**Discussion by mingyangpeng** [isl-org/Open3D#3724] [Q&A]
Hello, I need help that I want to see the change state of point cloud before and after processing, how to keep the previous window open and create a new window. I didn't find any example.

**Comment by ssheorey** [comment]
Hi @mingyangpeng  check out this example for visualizing updating point clouds in multiple windows:

https://github.com/intel-isl/Open3D/blob/master/examples/python/gui/multiple-windows.py

**Reply by mingyangpeng** [reply]
Thanks for your advice, I try it OK with python，but when I  check out this example with c++ code,it showed below

erminate called after throwing an instance of 'std::runtime_error'
  what():  [Open3D Error] (static void open3d::visualization::rendering::EngineInstance::SetResourcePath(const string&)) /home/jhon/dev/Open3D/cpp/open3d/visualization/rendering/filament/FilamentEngine.cpp:66: Can't find resource directory: /home/jhon/kdnv_nano/../resources


gui::Application::GetInstance().Initialize();    <--- I find initial with error.
Is there a simple demo with C++, split window or keep multi window open

**Reply by Kristijan10048** [reply]
I have the same issue https://github.com/isl-org/Open3D/discussions/3833. I am recompiling with hardcoded value to my local resources  folder and see if this will help. The  problematic function is FindResourcePath(int argc, const char *argv[])  in Open3D\cpp\open3d\visualization\gui\Application.cpp

**Comment by germanros1987** [comment]
@errissa could you take a look at this? Is this feature still working?

**Comment by errissa** [comment]
@mingyangpeng I confirmed that the `MultipleWindows` C++ example (`examples/cpp/MultipleWindows.cpp`) works as expected.

The visualizer is a little picky about where it looks for the resources it requires so that's why you got the error you got. You have to run the `MultipleWindows` example from the examples build directory: `<cmake_build_dir>/bin/examples`. In general, the visualizer looks for its resources in `<current_directory>/resources` and if it doesn't find it there, then `../resources`, and finally `current_directory/share/resources`.
